### PR TITLE
DRAFT: SC4-1-2-role-has-required-states-and-properties

### PR DESCRIPTION
**Pull request by annethyme**
Created from issue #155

# Checklist for Pull Request
🚨Please review the [guidelines for contributing](https://auto-wcag.github.io/auto-wcag/pages/contribute.html) to this repository.

- [X] Make sure you are requesting to **pull a issue/feature/bugfix branch** (right side). Don't request your master branch!
- [X] Make sure you are making a pull request against the **master branch** (left side). Also you should start *your branch* off *master branch*.
- [X] Kindly prefix the pull request name with `DRAFT` or `FINAL` accordingly. For more details, read [Rule Design Guidelines](https://auto-wcag.github.io/auto-wcag/pages/design/review.html)
- [X] Provide a reference to the issue being addressed.

#Reviewing Guidelines
- For `DRAFT` PR's, leave any comments you have and approve if you’ve reviewed the entire pull request. 
- Once there have been three such reviews the pull request can be merged. 
- Any comments left will be taken up in an updated pull request, which will be necessary before a FINAL release”.
- For `FINAL` PR's, review and approve changes to be merged.

# How to Review And Approve
- Go to the “files changed” tab, there you will have the option to leave comments on different lines. 
- Once the review is completed, find the “Review changes” button in the top right, select “approve” and click “Submit review”

### Description
Please describe your pull request.

💔Thank you!

### CODE REVIEW

File: _drafts/SC4-1-2-role-has-required-states-and-properties.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+Any HTML or SVG element that is [exposed to assistive technologies](#exposed-to-assistive-technologies) and has a [semantic role](#semantic-role). 
+
+### Expectation 1
+
+For each test target all [required states and properties](https://www.w3.org/TR/wai-aria/#requiredState) for that role as specified by [WAI-ARIA](https://www.w3.org/TR/wai-aria) are present.
+
+### Expectation 2
+
+Each of the [required states and properties](https://www.w3.org/TR/wai-aria/#requiredState) have a valid value according to the [value type](https://www.w3.org/TR/wai-aria-1.1/#propcharacteristic_value) for the given state or property as specified by [WAI-ARIA](https://www.w3.org/TR/wai-aria).
```
Comment (jkodu) [inline _drafts/SC4-1-2-role-has-required-states-and-properties.md]: [required states and properties]
- I see no definition for this. It should be a part of this PR?

File: _drafts/SC4-1-2-role-has-required-states-and-properties.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+**Note:** 
+- If the value of the required states and properties is of value type `ID Reference` or `ID Reference List` ensure that an element with the given id exists in the document.
```
Comment (jkodu) [inline _drafts/SC4-1-2-role-has-required-states-and-properties.md]: `ID Reference` or `ID Reference List`, these could do with being anchors to what they mean?

File: _drafts/SC4-1-2-role-has-required-states-and-properties.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+For each test target all [required states and properties](https://www.w3.org/TR/wai-aria/#requiredState) for that role as specified by [WAI-ARIA](https://www.w3.org/TR/wai-aria) are present.
+
+### Expectation 2
+
+Each of the [required states and properties](https://www.w3.org/TR/wai-aria/#requiredState) have a valid value according to the [value type](https://www.w3.org/TR/wai-aria-1.1/#propcharacteristic_value) for the given state or property as specified by [WAI-ARIA](https://www.w3.org/TR/wai-aria).
+
+**Note:** 
+- If the value of the required states and properties is of value type `ID Reference` or `ID Reference List` ensure that an element with the given id exists in the document. 
+- If the value of the required states and properties is of value type `URI` ensure that it matches the [generic URI syntax](https://www.ietf.org/rfc/rfc3986.txt).
```
Comment (jkodu) [inline _drafts/SC4-1-2-role-has-required-states-and-properties.md]: [generic URI syntax] - we decided to condense the meaning of the word generic URI syntax?

File: _drafts/SC4-1-2-role-has-required-states-and-properties.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+## Assumptions
+
+- Even though browsers and/or assistive technologies seem to support leaving out at least some [required states and properties](https://www.w3.org/TR/wai-aria-1.1/#requiredState) if an [implicit value for role](https://www.w3.org/TR/wai-aria-1.1/#implictValueForRole) is specified in [WAI-ARIA](https://www.w3.org/TR/wai-aria-1.1/#implictValueForRole), we do not consider it enough to pass this rule, that WAI-ARIA specifies an implicit value for the required states or properties for a role.
```
Comment (ShadowBB) [inline _drafts/SC4-1-2-role-has-required-states-and-properties.md]: This read like a note. I think the assumption itself could be slightly more concise. Something along the lines of "leaving out [required states and properties](https://www.w3.org/TR/wai-aria-1.1/#requiredState) if an [implicit value for role](https://www.w3.org/TR/wai-aria-1.1/#implictValueForRole) is specified in [WAI-ARIA](https://www.w3.org/TR/wai-aria-1.1/#implictValueForRole) is not fully accessibility supported."

Comment (jeankaplansky) [reply _drafts/SC4-1-2-role-has-required-states-and-properties.md]: +1 @ShadowBB

Comment (jeankaplansky) [reply _drafts/SC4-1-2-role-has-required-states-and-properties.md]: +1 @ShadowBB

**Pull request merged**